### PR TITLE
update readme files

### PR DESCRIPTION
**Pull request by CunliangGeng**
Updated the information of project markdown files

Major changes
- updated README.md
- updated README.dev.md
- updated project_setup.md
- deleted 03_readthedocs.md
- updated install-nplinker-deps

### CODE REVIEW

**Comment by CunliangGeng** [comment]
* **#229** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/229?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#228** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/228?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#227** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/227?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#226** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/226?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#224** <a href="https://app.graphite.dev/github/pr/NPLinker/nplinker/224?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @CunliangGeng and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: project_setup.md [hunk 1]
```diff
@@ -43,21 +30,15 @@ help you decide which tool to use for packaging.
 - The testing framework used is [PyTest](https://pytest.org)
   - [PyTest introduction](https://pythontest.com/pytest-book/)
   - PyTest is listed as a development dependency
-  - This is configured in `setup.cfg`
+  - This is configured in `pyproject.toml`
```
Comment (gcroci2) [inline project_setup.md:33]: This `project_setup.md` file is new to me. What does it add to the README?

**Review by gcroci2 — APPROVED** [review]

File: project_setup.md (hunk 1 above)
Comment (CunliangGeng) [reply project_setup.md:33]: Good question. This file is taken from the python template of eScience center, which helps on setting up the repo. I don't think it is very necessary if developers are already familiar with github and python repos, but it's nice for new developers.

**Comment by CunliangGeng** [comment]
### Merge activity

* **Mar 22, 9:30 AM EDT**: @CunliangGeng started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/228).
* **Mar 22, 9:31 AM EDT**: @CunliangGeng merged this pull request with [Graphite](https://app.graphite.dev/github/pr/NPLinker/nplinker/228).

**Pull request merged**